ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/style/ImageComponentStyle.kt

```diff
@@ -0,0 +1,83 @@
+package com.revenuecat.purchases.ui.revenuecatui.components.style
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Density
+import com.revenuecat.purchases.paywalls.components.properties.ImageUrls
+import com.revenuecat.purchases.paywalls.components.properties.Size
+import com.revenuecat.purchases.paywalls.components.properties.SizeConstraint.Fill
+import com.revenuecat.purchases.paywalls.components.properties.SizeConstraint.Fit
+import com.revenuecat.purchases.paywalls.components.properties.SizeConstraint.Fixed
+import com.revenuecat.purchases.paywalls.components.properties.ThemeImageUrls
+import com.revenuecat.purchases.ui.revenuecatui.components.ktx.urlsForCurrentTheme
+import com.revenuecat.purchases.ui.revenuecatui.components.properties.ColorStyle
+
+@Suppress("LongParameterList")
+@Immutable
+internal class ImageComponentStyle private constructor(
+    @get:JvmSynthetic
+    val visible: Boolean,
+    @get:JvmSynthetic
+    val themeImageUrls: ThemeImageUrls,
+    @get:JvmSynthetic
+    val size: Size,
+    @get:JvmSynthetic
+    val shape: Shape?,
+    @get:JvmSynthetic
+    val overlay: ColorStyle?,
+    @get:JvmSynthetic
+    val contentScale: ContentScale,
+) : ComponentStyle {
+
+    companion object {
+
+        @Suppress("LongParameterList")
+        @JvmSynthetic
+        @Composable
+        operator fun invoke(
+            visible: Boolean,
+            size: Size,
+            themeImageUrls: ThemeImageUrls,
+            shape: Shape?,
+            overlay: ColorStyle?,
+            contentScale: ContentScale,
+        ): ImageComponentStyle {
+            return ImageComponentStyle(
+                visible = visible,
+                size = size,
+                themeImageUrls = themeImageUrls,
+                shape = shape,
+                overlay = overlay,
+                contentScale = contentScale,
+            )
+        }
+    }
+
+    @JvmSynthetic
+    @ReadOnlyComposable
+    @Composable
+    fun adjustedSize(): Size {
```

Comment:
I moved this to the `ImageComponentStyle` since it felt like a better place to put it than as simple extensions of the `Size` class, since here is where we actually have the data to adjust the size. Not sure if there are other components that would require a similar adjustment, in which case, I think we can extract it to extension functions as suggested. Wdyt?

Reply:
I like this!